drivers/core_demographics_report/app/views/core_demographics_report/warehouse_reports/core/_ages.haml

```diff
@@ -26,16 +26,16 @@
           %tbody
             %tr
               %th All
-              %td= number_with_delimiter report.adult_count
+              %td= link_to_if can_view_clients?, number_with_delimiter(report.adult_count), details_core_demographics_report_warehouse_reports_core_index_path(report.filter.for_params.deep_merge(key: :adult_scope, filters: { start: report.filter.start_date, end: report.filter.end_date })), target: :_blank
```

Comment:
Maybe DRY this up with a helper? Except for the key, the paths are the same, yes?

Reply:
Yeah, that was really tempting.  I think it can be done now, but took a few iterations to get to a point where that would make sense.